src/conditions/datadog_search.rs

```diff
@@ -27,215 +26,297 @@ impl_generate_config_from_default!(DatadogSearchConfig);
+            field,
+            lower,
+            lower_inclusive,
+            upper,
+            upper_inclusive,
+        } => {
+            match (&lower, &upper) {
+                // If both bounds are wildcards, just check that the field exists to catch the
+                // special case for "tags".
+                (ComparisonValue::Unbounded, ComparisonValue::Unbounded) => exists(field, log),
+                // Unbounded lower.
+                (ComparisonValue::Unbounded, _) => {
+                    let op = if *upper_inclusive {
+                        Comparison::Lte
+                    } else {
+                        Comparison::Lt
+                    };
+                    compare(field, op, upper, log)
+                }
+                // Unbounded upper.
+                (_, ComparisonValue::Unbounded) => {
+                    let op = if *lower_inclusive {
+                        Comparison::Gte
+                    } else {
+                        Comparison::Gt
+                    };
 
-                array_match(field, move |values| {
-                    values.contains(&Value::Bytes(Bytes::copy_from_slice(to_match.as_bytes())))
-                })
-            }
-            // Individual tags are compared by element key:value.
-            Field::Tag(tag) => {
-                let value_bytes = Value::Bytes(format!("{}:{}", tag, to_match).into());
+                    compare(field, op, lower, log)
+                }
+                // Definitive range.
+                _ => {
+                    let lower_op = if *lower_inclusive {
+                        Comparison::Gte
+                    } else {
+                        Comparison::Gt
+                    };
 
-                array_match("tags", move |values| values.contains(&value_bytes))
-            }
-            // Everything else is matched by string equality.
-            Field::Reserved(field) | Field::Facet(field) => {
-                let to_match = to_match.to_owned();
+                    let upper_op = if *upper_inclusive {
+                        Comparison::Lte
+                    } else {
+                        Comparison::Lt
+                    };
 
-                string_match(field, move |value| value == to_match)
+                    compare(field, lower_op, lower, log) && compare(field, upper_op, upper, log)
+                }
             }
         }
+        Op::Not(matcher) => !EventFilter::run(matcher, log),
+        Op::Nested(matcher) => EventFilter::run(matcher, log),
+    }
+}
+
+fn exists(field: &Field, log: &LogEvent) -> bool {
+    match field {
+        Field::Tag(tag) => match log.get("tags") {
+            Some(Value::Array(values)) => values
+                .iter()
+                .filter_map(|value| {
+                    if let Value::Bytes(bytes) = value {
+                        std::str::from_utf8(bytes).ok()
+                    } else {
+                        None
+                    }
```

Comment:
`match` again?

Reply:
The if let idiom looks pretty common in `datadog_search.rs`. I could flip them all if you feel strongly about it.